### PR TITLE
Make driver speed not cause problems with driving

### LINKED ISSUE #50462: Less fumbling with vehicles
# Is your feature request related to a problem? Please describe.
When turning a moving vehicle, the character has a high chance to fumble with the controls, causing them to turn randomly or not turn at all.  My current character is at vehicles 7, and while driving a pristine motorcycle at 20 MPH, I fumble at least a dozen times. Even as slow as 4 MPH I've fumbled up to 10 times. This seems wildly unrealistic, and it almost makes driving worse than simply walking. 

# Describe the solution you'd like
I think that fumbling should be greatly reduced, and outright stopped at a certain vehicles skill. 

# Describe alternatives you've considered
Bigger vehicles could be locked until you reach a certain level in the vehicles skill, as in real life larger vehicles require training beyond simple driver's education to attain licenses. The vehicles skill could also/instead play a bigger role in how much focus is drained when driving. 

# Additional context
EDIT: It seems that increasing the player's speed via magiclysm items such as a ring of speed cause the fumbling, and driving works as intended without them. This should probably be relabeled to a bug, since I don't think the increased speed is meant to cause fumbling.

### PR DESCRIPTION
**Pull request by actual-nh**
#### Summary
None

#### Purpose of change

A high driver speed, such as from a Magiclysm haste spell, was causing increased - not decreased - fumbles. Fixes #50462.

#### Describe the solution

400 was being compared to a cost that was partially determined by driver speed; instead, compare to `std::max( driver.get_speed() * 4, 400 )`.

#### Testing

CI; would appreciate playtesting by people with familiarity with vehicles in CDDA.

### CODE REVIEW

**Comment by eltank** [comment]
Personally, I would lean towards hard-capping the number of steering actions to 3 per turn regardless of player speed.

That means changing the line above:
```
cost = std::max( driver.get_speed(), 100 ) * ( 1.0f - ( -penalty / 10.0f ) * 2 / 3 );
```
to
```
cost = 100  * ( 1.0f - ( -penalty / 10.0f ) * 2 / 3 );
```

File: src/vehicle_move.cpp [hunk 1]
```diff
@@ -1398,7 +1398,7 @@ void vehicle::pldrive( Character &driver, const point &p, int z )
             cost = std::max( driver.get_speed(), 100 ) * ( 1.0f - ( -penalty / 10.0f ) * 2 / 3 );
         }
 
-        if( penalty > skill || cost > 400 ) {
+        if( penalty > skill || cost > std::max( driver.get_speed() * 4, 400 ) ) {
```
Comment (eltank) [inline src/vehicle_move.cpp:1401]: I would write it as:
```suggestion
        if( penalty > skill || ( penalty > 0 && cost > std::max( driver.get_speed() * 4, 400 ) ) ) {
```
Basically, we should never fumble if the penalty is less than 0 (meaning we have more than enough skill).

**Review by eltank — CHANGES_REQUESTED** [review]

**Comment by eltank** [comment]
I'll update soon with results of playtesting the latest code change.

**Comment by eltank** [comment]
> I'll update soon with results of playtesting the latest code change.

Drove a motorcycle, mojocycle and SWAT truck at 60mph while hasted to 631 player speed with no fumbling. I had debug code printing the penalty and skill at every turn. Skill was capped at 10 (due to high stats and vehicles skill) and penalty was always negative.

**Review by eltank — APPROVED** [review]

**Comment by mqrause** [comment]
> ```
> cost = std::max( driver.get_speed(), 100 ) * ( 1.0f - ( -penalty / 10.0f ) * 2 / 3 );
> ```

Not relevant to the PR, but what's with the calculation there? Why not shorten the second part to `(1.0f + penalty / 15.0f)` and add a comment what it means?

**Comment by eltank** [comment]
> > ```
> > cost = std::max( driver.get_speed(), 100 ) * ( 1.0f - ( -penalty / 10.0f ) * 2 / 3 );
> > ```
> 
> Not relevant to the PR, but what's with the calculation there? Why not shorten the second part to `(1.0f + penalty / 15.0f)` and add a comment what it means?

Due to skill being capped at 10, penalty can't go below -10, so this "/ 10" is normalizing it to [0,1], which makes it clearer that it gives at most a 2/3 reduction in cost.
I think @mlangsdorf would say that this magic number 10 needs to be defined as a constant (shared with the skill capping code above) and we shouldn't introduce more magic numbers (such as the 15 you suggested).

**Comment by mqrause** [comment]
Sure, I just don't think the way it's written makes it clearer what it's doing than a comment and a magic number.

**Comment by eltank** [comment]
> Sure, I just don't think the way it's written makes it clearer what it's doing than a comment and a magic number.

I will have reason to mess with this code later (need to refactor the cost calculation outside of pldrive in order to improve autodrive) and I'll try to clean it up then.

**Pull request merged**